// vitepress-theme-teek/es/components/theme/HomeFriendLinkCard/src/ItemInfo.vue2.mjs
import { defineComponent, useTemplateRef, onMounted, openBlock, createElementBlock, unref, createElementVNode, normalizeClass, toDisplayString } from 'vue';
import { withBase } from 'vitepress';

const _hoisted_1 = ["href", "aria-label"];
const _hoisted_2 = { class: "friend-avatar skeleton-image" };
const _hoisted_3 = ["src", "alt"];
const _hoisted_4 = ["title"];
const _hoisted_5 = ["title"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ItemInfo",
  props: {
    item: {},
    ns: {}
  },
  setup(__props) {
    const imgInstance = useTemplateRef("imgInstance");
    const addLoadedClass = (e) => {
      if (e?.target) e.target.classList.add("loaded");
      else if (imgInstance.value) imgInstance.value.classList.add("loaded");
    };
    onMounted(() => {
      const img = imgInstance.value;
      if (!img) return;
      if (img.complete) addLoadedClass();
      else {
        img.addEventListener("load", () => addLoadedClass());
        img.addEventListener("error", () => addLoadedClass());
      }
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("a", {
        href: __props.item.link && unref(withBase)(__props.item.link),
        target: "_blank",
        class: "hover-color flx-align-center",
        "aria-label": __props.item.name
      }, [
        createElementVNode("div", _hoisted_2, [
          createElementVNode("img", {
            ref_key: "imgInstance",
            ref: imgInstance,
            src: __props.item.avatar && unref(withBase)(__props.item.avatar),
            class: "avatar",
            alt: __props.item.alt || __props.item.name,
            "aria-hidden": "true",
            onLoad: addLoadedClass,
            onError: addLoadedClass
          }, null, 40, _hoisted_3)
        ]),
        createElementVNode(
          "div",
          {
            class: normalizeClass(__props.ns.e("list__item__info"))
          },
          [
            createElementVNode("div", {
              class: "friend-name sle",
              title: __props.item.name
            }, toDisplayString(__props.item.name), 9, _hoisted_4),
            createElementVNode("div", {
              class: "friend-desc sle",
              title: __props.item.desc
            }, toDisplayString(__props.item.desc), 9, _hoisted_5)
          ],
          2
          /* CLASS */
        )
      ], 8, _hoisted_1);
    };
  }
});

export { _sfc_main as default };
